app/controllers/results_controller.rb

```diff
@@ -336,13 +347,22 @@ def codeviewer
     @assignment = Assignment.find(params[:assignment_id])
     @submission_file_id = params[:submission_file_id]
     @focus_line = params[:focus_line]
-
+    @grouping = @current_user.grouping_for(Integer(params[:assignment_id]))
     @file = SubmissionFile.find(@submission_file_id)
     @result = @file.submission.get_latest_result
-    # Is the current user a student?
+
+    if @current_user.is_a_reviewer?(@assignment)
+      prs = PeerReview.where(reviewer_id: @grouping.id)
```

Comment:
In general, you should be using associations a lot more. It's very rare to need to do a query of the form `Class.where...`, but instead something like `@grouping.peer_reviews`.